internal/querynodev2/segments/segment_loader.go

```diff
@@ -923,6 +926,36 @@ func (loader *segmentLoader) getFieldType(segment *LocalSegment, fieldID int64)
 	return 0, merr.WrapErrFieldNotFound(fieldID)
 }
 
+func (loader *segmentLoader) LoadIndex(ctx context.Context, segment *LocalSegment, loadInfo *querypb.SegmentLoadInfo) error {
+	log := log.Ctx(ctx).With(
+		zap.Int64("collection", segment.Collection()),
+		zap.Int64("segment", segment.ID()),
+	)
+
+	log.Info("segment loader start to load index")
```

Comment:
1. Need to avoid multiple requests loading index for the same index
2. Need to check whether the memory is enough to load index
Ref `Load()` method to fix these

Reply:
For now, the load index memory overhead factor is 2x